client.go

```diff
@@ -1280,8 +1299,17 @@ func (f *File) Chmod(mode os.FileMode) error {
 // that if the size is less than its current size it will be truncated to fit,
 // the SFTP protocol does not specify what behavior the server should do when setting
 // size greater than the current size.
+// We send a SSH_FXP_FSETSTAT here since we have a file handle
 func (f *File) Truncate(size int64) error {
-	return f.c.Truncate(f.path, size)
+	err := f.c.setfstat(f.handle, sshFileXferAttrSize, uint64(size))
+	if err == nil {
```

Comment:
Note that [`os.File.Truncate`](https://golang.org/pkg/os/#File.Truncate) says: “It does not change the I/O offset.”

An in fact [`man ftruncate`](https://linux.die.net/man/2/ftruncate) also says: “The file offset is not changed.”

Additional information about how to handle “what if we’re past the end of the new file length before a `truncate`? Available here: https://stackoverflow.com/questions/60950996/how-file-offset-of-read-or-write-change-when-file-is-truncated-to-zero-by-ot

Note: a `Read()` when `offset > length` should return zero-bytes read. A `Write()` when `offset > length` should ftruncate to the new offset, before extending further with the write.

Reply:
Oops I missed that thank you. 

The `Read()` and `Write()` test cases are client side tests I didn't include them